### PR TITLE
Mm 38590 allow disabling retrospective

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This pull request will introduce a per-playbook configuration that allows disabling the retrospective functionality for subsequent runs created using that playbook.

- Added **Enable retrospective** toggle to playbook edit form. When disabled, **Retrospective reminder interval** and **Retrospective template** settings are disabled.

https://user-images.githubusercontent.com/4368372/144270337-6b814229-1275-413c-ad3e-864b59fb1215.mp4

- When a run is finished, no prompt occurs after finishing a run to start a retrospective

https://user-images.githubusercontent.com/4368372/144270430-6133d20a-2bfc-4423-a202-0cdd926b8b9e.mp4

- When viewing the playbook run details, the retrospective field should be replaced with a message indicating, “Retrospectives were disabled for this playbook run.”

https://user-images.githubusercontent.com/4368372/144270507-7977d2c4-5fe7-4914-9277-3e7acb50adb9.mp4

- When viewing playbook preview, a retrospective section is omitted

https://user-images.githubusercontent.com/4368372/144270556-2dc29bca-87d2-4c24-8aab-32f55ddb8b3d.mp4

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-38590

#### Checklist

- ~~[ ] Telemetry updated~~
- ~~[ ] Gated by experimental feature flag~~
- ~~[ ] Unit tests updated~~

### CODE REVIEW

**Comment by cpoile** [comment]
@abhijit-singh For the video above labelled `When viewing playbook preview, a retrospective section is omitted` -- I was chatting with @gbochora and we weren't sure if maybe you wanted the section to be there but to say something like "The retrospective step has been disabled" -- so that users might realize that that's a thing and then go and turn it on. 

Let us know, thanks!

**Comment by abhijit-singh** [comment]
@cpoile @gbochora  That's a good idea but I think it is okay to completely hide retrospective from the preview page for now as not everyone viewing that page might have permission to edit the playbook and the playbook admin will have intentionally disabled it at some point.

This could also be done as a separate effort after we have the enable/disable functionality in place if feel it'll add significant value.

File: tests-e2e/cypress/integration/channels/actions_spec.js [hunk 1]
```diff
@@ -597,4 +597,44 @@ describe('channels > actions', () => {
             });
         });
     });
+
+    describe('when a playbook run is finished', () => {
```
Comment (cpoile) [inline tests-e2e/cypress/integration/channels/actions_spec.js:601]: again, this is awesome work on the e2e tests! 👍

File: webapp/src/components/backstage/playbooks/playbook_preview_retrospective.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const PlaybookPreviewRetrospective = (props: Props) => {
                             {props.playbook.retrospective_reminder_interval_seconds === 0 ? 'ONCE' : formatDuration(Duration.fromObject({seconds: props.playbook.retrospective_reminder_interval_seconds}), 'long')}
                         </TextBadge>
                     )}
-                    enabled={true}
+                    enabled={props.playbook.retrospective_enabled}
```
Comment (cpoile) [inline webapp/src/components/backstage/playbooks/playbook_preview_retrospective.tsx:52]: I think this was the bit we can remove, given the response from Abhijit?

**Review by cpoile — APPROVED** [review]
Again, awesome work @gbochora!
Just one small change, and then a little conflict with master. Otherwise I'll send it to @agarciamontoro for a second look.

File: webapp/src/components/backstage/playbooks/playbook_preview_retrospective.tsx (hunk 2 above)
Comment (gbochora) [reply webapp/src/components/backstage/playbooks/playbook_preview_retrospective.tsx]: Yes, that's correct. I've removed and pushed changes.

File: server/app/playbook_run.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ const (
 
 // PlaybookRun holds the detailed information of a playbook run.
 //
-// NOTE: When adding a column to the db, search for "When adding a Playbook Run column" to see where
+// NOTE: When adding a column to the db, search for "When adding a PlaybookRun column" to see where
```
Comment (agarciamontoro) [inline server/app/playbook_run.go:21]: Good catch :)

File: tests-e2e/cypress/integration/runs/details_spec.js [hunk 4]
```diff
@@ -108,4 +108,43 @@ describe('runs > details', () => {
...
+                    playbookRunName: name,
+                    ownerUserId: testUser.id,
+                });
+            }).then((run) => {
+                playbookRun = run
+            })
+        });
+
+
+        it('should shows, that retrospectives were disabled message', () => {
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/runs/details_spec.js:139]: Typo
```suggestion
        it('should show the retrospectives were disabled message', () => {
```

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx [hunk 5]
```diff
@@ -54,3 +59,12 @@ export const Retrospective = (props: Props) => {
         </Container>
     );
 };
+
+const RetrospectiveDisabledText = styled.h2`
+    font-family: Open Sans;
+    font-style: normal;
+    font-weight: normal;
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx:66]: Can we make this a simple div? I think if we do that, we won't need these font rules.

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Great work again! There's just one tiny CSS thing below, but otherwise looks fantastic, great job :tada:

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx (hunk 5 above)
Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx:66]: You mean like this:
```
const RetrospectiveDisabledText = styled.div`
    font-size: 20px;
    color: var(--center-channel-color);
    text-align: left;
`;
```

Comment (agarciamontoro) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx:66]: Exactly, @gbochora! But I have not tested it, so I'm not sure it's exactly the same. The only concern I have with the current code is that the rules
```css
font-family: Open Sans;
font-style: normal;
font-weight: normal; 
```
are the default ones (probably overridden by the h2 style, and that's why you need them), so we should not need it.

Comment (gbochora) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx]: I've tested, and the text is a little bolder without `font-weight: normal;`, so I've decided to leave this parameter. The rest is fine.

**Comment by gbochora** [comment]
@agarciamontoro I've merged with master and pushed changes.

**Review by agarciamontoro — APPROVED** [review]
Great, @gbochora, thank you for all your work here :tada:

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx (hunk 5 above)
Comment (agarciamontoro) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx]: That's perfect, thanks for looking into this!

**Pull request merged**